### ISSUE TITLE
linter: `eslint/no-lone-blocks` flags empty loops

### ISSUE THREAD

**Issue by jeanne-cai** [oxc-project/oxc#22648]
### What version of Oxlint are you using?

1.66.0

### What command did you run?

`yarn oxlint -c oxlint.config.mts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
    categories: {
        correctness: 'off',
    },
    rules: {
        'eslint/no-lone-blocks': 'error',
    },
});
```

### What happened?

The rule `eslint/no-lone-blocks` reports empty loops as lone blocks. ESLint does not report empty loops for this rule.

Example:
```ts
// lone block
{
}

// not a lone block
let i = 5;
while (i > 0) {}

// not a lone block
for (let idx = 0; idx < 5; idx++) {}
```

Output:

```bash
  × eslint(no-lone-blocks): Block is unnecessary.
   ╭─[test.ts:2:1]
 1 │     // lone block
 2 │ ╭─▶ {
 3 │ ╰─▶ }
 4 │     
   ╰────
  help: Remove the unnecessary block statement. If you need to limit variable scope, consider using a function or module instead.

  × eslint(no-lone-blocks): Block is unnecessary.
   ╭─[test.ts:7:15]
 6 │ let i = 5;
 7 │ while (i > 0) {}
   ·               ──
 8 │ 
   ╰────
  help: Remove the unnecessary block statement. If you need to limit variable scope, consider using a function or module instead.

  × eslint(no-lone-blocks): Block is unnecessary.
    ╭─[test.ts:10:35]
  9 │ // not a lone block
 10 │ for (let idx = 0; idx < 5; idx++) {}
    ·                                   ──
    ╰────
  help: Remove the unnecessary block statement. If you need to limit variable scope, consider using a function or module instead.

Found 0 warnings and 3 errors.
Finished in 293ms on 1 file with 1 rules using 16 threads.
```
